hr_timesheet_activity_begin_end_readonly_duration/models/analytic_account.py

```diff
@@ -0,0 +1,19 @@
+# © 2023 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3.0 (https://www.gnu.org/licenses/lgpl-3.0.html)
+
+from odoo import fields, models, api
+from datetime import timedelta
+
+
+class AccountAnalyticLine(models.Model):
+    _inherit = "account.analytic.line"
+
+    unit_amount = fields.Float(string="Quantity", default=0.0, readonly=True)
+
+    @api.constrains("time_start", "time_stop")
+    def _check_time_start_stop(self):
+        for line in self:
+            start = timedelta(hours=self.time_start)
+            stop = timedelta(hours=self.time_stop)
+            line.unit_amount = (stop - start).seconds / 3600
```

Comment:
Why is this code needed? If it's finally required, better try this (code reuse, from https://github.com/OCA/timesheet/blob/14.0/hr_timesheet_activity_begin_end/models/account_analytic_line.py#L72):
```suggestion
            line.onchange_hours_start_stop()
```